target/linux/ar71xx/base-files/lib/upgrade/platform.sh

```diff
@@ -40,25 +40,72 @@ platform_find_kernelpart() {
 	done
 }
 
+platform_find_rootfspart() {
+	local part
+	for part in "${1%:*}" "${1#*:}"; do
+		[ "$part" != "$2" ] && echo "$part"; break
+	done
+}
+
+platform_do_upgrade_combined_mikrotik_helper(){
+	local kernelpart="$2"
+	local kern_blocks_with_info_hdr="$3" #kernel image info_header size in blocks + kernel body size in blocks
```

Comment:
Minor stylistic remark: personally I usually put a space between `#` and the text after it, makes it less likely to be confused with commented-out commands